gene2phenotype_project/gene2phenotype_app/views/search.py

```diff
@@ -134,47 +141,86 @@ def get_queryset(self):
 
             if not queryset.exists():
                 self.handle_no_permission('g2p_id', search_query)
-
+        
+        elif search_type == 'draft' and user.is_authenticated:
+            queryset = CurationData.objects.filter(
+                gene_symbol=search_query
+                ).order_by('stable_id__stable_id').distinct()
+            
+            # to extend the queryset being annotated when it is draft,
+            # want to return username so curator can see who is curating
+            # adding the curator email, incase of the notification.
+            queryset = queryset.annotate(user_name=F('user_id__username'), user_email=F('user__email'))
+
+            
+            if not queryset.exists():
+                self.handle_no_permission("draft", search_query)
+    
         else:
             self.handle_no_permission('Search type is not valid', None)
-
+        
         new_queryset = []
         if queryset.exists():
-            for lgd in queryset:
+            if search_type != 'draft':
+                for lgd in queryset:
                 # If the user is not logged in, only show visible panels and remove disputed and refuted records
-                if user.is_authenticated == False:
-                    lgdpanel_select = LGDPanel.objects.filter(
-                        Q(lgd=lgd, panel__is_visible=1, is_deleted=0)
-                        & ~Q(lgd__confidence__value='disputed') & ~Q(lgd__confidence__value='refuted'))
-                else:
-                    lgdpanel_select = LGDPanel.objects.filter(lgd=lgd, is_deleted=0)
-
-                lgd_panels = []
-                for lp in lgdpanel_select:
-                    lgd_panels.append(lp.panel.name)
-
-                # Add new property to LGD object
-                lgd.panels = lgd_panels
-                if lgd_panels:
-                    new_queryset.append(lgd)
+                    if user.is_authenticated is False:
+                        lgdpanel_select = LGDPanel.objects.filter(
+                            Q(lgd=lgd, panel__is_visible=1, is_deleted=0)
+                            & ~Q(lgd__confidence__value='disputed') & ~Q(lgd__confidence__value='refuted'))
+                    else:
+                        lgdpanel_select = LGDPanel.objects.filter(lgd=lgd, is_deleted=0)
+
+                    lgd_panels = []
+                    for lp in lgdpanel_select:
+                        lgd_panels.append(lp.panel.name)
+
+                    # Add new property to LGD object
+                    lgd.panels = lgd_panels
+                    if lgd_panels:
+                        new_queryset.append(lgd)
+            else:
+                return queryset
 
         return new_queryset
 
     def list(self, request, *args, **kwargs):
         queryset = self.get_queryset()
+        serializer = self.get_serializer_class()
+
+        
         list_output = []
+        if serializer == "LocusGenotypeDisease.Serializer":
```

Comment:
This is not working, when I search for type=gene the endpoint fails.
You should update this line to:
`if issubclass(serializer, LocusGenotypeDiseaseSerializer):`